src/endpoint/s3/ops/s3_put_object.js

```diff
@@ -65,6 +65,8 @@ async function put_object(req, res) {
     }
     s3_utils.set_encryption_response_headers(req, res, reply.encryption);
 
+    res.size_for_notif = size || reply.size;
```

Comment:
Under which circumstances the size would be 0 and reply.size would be non zero?
The opposite is possible but if we have called putobject to upload some data then "size"sould be there if not then reply.size should also be zero. 
It is possible that the call to upload was not successful so reply.size might be less than size or zero.

Reply:
size is zero (and reply.size is not zero) when copy_source is true.